crates/top/rerun/src/commands/entrypoint.rs

```diff
@@ -1309,3 +1246,133 @@ fn stream_to_rrd_on_disk(
 
     Ok(())
 }
+
+/// Describes how to handle URLs passed on the CLI.
+struct UrlParamProcessingConfig {
+    data_sources_from_http_urls: bool,
+    data_sources_from_redap_datasets: bool,
+    data_source_from_filepaths: bool,
+}
```

Comment:
I suppose this embodies "what doesn't right" by you 😅  Can't say I'm in love either, but at least this is very explicit at what its doing.